src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c

```diff
@@ -22,99 +98,1072 @@ const ActorInit En_Sellnuts_InitVars = {
+            this->unk_34C = 16;
+            this->collider.dim.height = 32;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_UP);
+            func_8013BC6C(&this->skelAnime, D_80ADD990, 16);
+        } else if (this->unk_34C == 17) {
+            if (DECR(this->unk_34E) == 0) {
+                this->unk_34E = Rand_ZeroOne() * 10.0f;
+                this->unk_34C = 2;
+                this->collider.dim.height = 32;
+                func_8013BC6C(&this->skelAnime, D_80ADD990, 2);
+            }
+        } else if (this->unk_34C == 16) {
+            if (DECR(this->unk_34E) == 0) {
+                this->unk_34E = Rand_S16Offset(40, 40);
+                this->unk_34C = 18;
+                this->collider.dim.height = 32;
+                func_8013BC6C(&this->skelAnime, D_80ADD990, 18);
+            }
+        }
+    }
+}
+
+void func_80ADB4F4(EnSellnuts* this, GlobalContext* globalCtx) {
+    if (this->skelAnime.animCurrentFrame == this->skelAnime.animFrameCount) {
+        this->actionFunc = func_80ADB544;
+        func_8013BC6C(&this->skelAnime, D_80ADD990, 0);
+    }
+}
+
+void func_80ADB544(EnSellnuts* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 3, 0x7D0, 0);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        if (func_800B8708(globalCtx) == 0x2A) {
+            player->actor.textId = D_80ADD928[this->unk_33A];
+            this->unk_340 = player->actor.textId;
+            this->actionFunc = func_80ADBAB8;
+        } else {
+            switch (this->unk_340) {
+                case 0x60E:
+                    gSaveContext.weekEventReg[17] |= 0x20;
+                    gSaveContext.weekEventReg[86] |= 4;
+                    func_801518B0(globalCtx, this->unk_340, &this->actor);
+                    this->actionFunc = func_80ADB0D8;
+                    break;
+
+                case 0x628:
+                    gSaveContext.weekEventReg[77] |= 0x40;
+                    gSaveContext.weekEventReg[86] |= 4;
+                    func_801518B0(globalCtx, this->unk_340, &this->actor);
+                    this->actionFunc = func_80ADB0D8;
+                    break;
+
+                case 0x614:
+                    gSaveContext.weekEventReg[17] |= 0x40;
+                    func_801518B0(globalCtx, this->unk_340, &this->actor);
+                    this->actionFunc = func_80ADB0D8;
+                    break;
+
+                case 0x610:
+                case 0x616:
+                case 0x629:
+                    func_801518B0(globalCtx, this->unk_340, &this->actor);
+                    this->actionFunc = func_80ADB0D8;
+                    break;
+
+                default:
+                    this->actionFunc = func_80ADBAB8;
+                    break;
+            }
+        }
+    } else {
+        goto label;
+    label: // fake
```

Comment:
Would this match without the `goto`?
```suggestion
    label: // fake
```